migration/idempotent/011-maintenance.sql

```diff
@@ -648,6 +648,122 @@ BEGIN
             EXECUTE _sql;
         END LOOP
         ;
+
+        -- Unfortunately, the following loop only really exists to make our dump/restore test pass.
+        -- When running `create extension promscale`, the span, link, and event tables are created
+        -- by the bootstrap superuser (typically postgres), and therefore the default privileges in
+        -- the `_timescaledb_internal` schema fire. This grants some privileges on the compressed
+        -- hypertables backing the span, link, and event tables to a couple of users.
+        -- On the other hand...
+        -- When restoring from a dump, the compressed hypertables backing the span, link, and event
+        -- tables are created directly by the user running the restore. In this case, the default
+        -- privileges do not fire. This loop effectively does what the default privileges would have
+        -- done.
+        FOR _sql IN
+        (
+            WITH x AS
+            (
+                SELECT
+                  h2.schema_name
+                , h2.table_name
+                FROM _timescaledb_catalog.hypertable h1
+                INNER JOIN _timescaledb_catalog.hypertable h2
+                ON (h1.schema_name = ANY(ARRAY['prom_data', '_ps_trace'])
+                AND h1.compressed_hypertable_id = h2.id)
+            )
+            SELECT format('GRANT ALL PRIVILEGES ON TABLE %I.%I TO prom_admin', x.schema_name, x.table_name)
+            FROM x
+            UNION ALL
+            SELECT format('ALTER TABLE %I.%I OWNER TO prom_admin', x.schema_name, x.table_name)
+            FROM x
+            UNION ALL
+            SELECT format('GRANT ALL PRIVILEGES ON TABLE %I.%I TO %I', x.schema_name, x.table_name, quote_ident(session_user))
```

Comment:
Using both `%I` and `quote_ident` will break whenever there is something to quote:
```SQL
select format('%I', quote_ident('&'));
┌─────────┐
│ format  │
├─────────┤
│ """&""" │
└─────────┘
(1 row)
```